Чипи Чипи Чапа Чапа

Deffris opened the pull request.

Commit:
SecurityFilterChain - > .anyRequest().permitAll());

made the basis of the request to create a task, some improvement is needed

Commit:
-add log Error
-refactor UserAuthController
-refactor TaskModel
-refactor DB Model(Sprints, TaskModel)
-refactor JwtUtil
-update db properties

Commit:
-add log Error
-refactor UserAuthController
-refactor TaskModel
-refactor DB Model(Sprints, TaskModel)
-refactor JwtUtil
-update db properties

MihailPribytkov commented on src/main/java/com/example/work_task/model/db/TaskModel.java:
```diff
@@ -4,41 +4,51 @@
+    @NonNull
     @Column
-    private Integer projectId;
+    private String projectId;
```
Я так понял у нас в БД projectId и sprintId = GUID, но у тебя в модели эти поля String. Можно для springId и projectId  использовать аннотацию @ManyToOne, которая позволяет работать с GUID

Mihail-Pribytkov commented on src/main/java/com/example/work_task/task/TaskController.java:
```diff
@@ -1,38 +1,131 @@
-    @GetMapping("/{id}")
-    public Optional<TaskModel> getTaskById(@PathVariable Integer id) {
-        return taskService.getTaskById(id);
+    private static final String ERROR_TITLE_FORMAT = "Некорректный формат поля TITLE";
+    private static final String ERROR_DESCRIPTION_FORMAT = "Некорректный формат поля DESCRIPTION";
+    private static final String ERROR_SPRINT_NOT_FOUND_OR_CLOSED = "Спринт закрыт или не существует";
+    private static final String ERROR_ESTIMATION_FORMAT = "Некорректный формат поля ESTIMATION";
+    private static final String ERROR_PRIORITY_VALUE = "Некорректное значение поля PRIORITY";
+    private static final String ERROR_TASK_TYPE_VALUE = "Некорректное значение поля TASK_TYPE";
+    private static final String ERROR_STATUS_VALUE = "Некорректное значение поля STATUS";
```
Может вынести ошибки в отдельный класс? Эти ошибки я думаю мы будем использовать не только в /create, но еще допустим в /update

MihailPribytkov commented on src/main/java/com/example/work_task/task/TaskController.java:
```diff
@@ -1,38 +1,131 @@
+        if (isValueInEnum(StatusName.class, taskModel.getStatus().toString())) {
+            errors.add(ERROR_STATUS_VALUE);
+        }
+
```
Тоже самое с валидацией. Не хотим вынести в отдельный класс? Может можно будет ее переиспользовать

Commit:
-delete taskExaple.json(Deprecated)
-rename com.example to ru.worktechlab
-add class ValidateTask
-update validateTask
-fix extractPayloadFromToken and getUserGuidFromJwtToken in JwtUtils
-update build.gradle(group=com.example to ru.worktechlab)

Commit:
-update SecurityConfig(set anyRequest().authenticated())
-fix RegisterDto
1.add @Data, delete setter/getter
2.add message for middleName

Commit:
-fix never used imports

Commit:
-fix create task

Commit:
-set token lifetime (20 min)
